Fix misc compiler warnings

jschmidt-icinga opened the pull request:
Together with #9411, #9728, #9729 and #9730 this fixes most of the remaining compiler warnings, except the following:

+ A few `-Wdangling-reference` in `scheduleddowntime-apply.cpp`, which I'm pretty sure are false positives.
+ The OpenSSL 3.0 `-Wdeprecated-declarations` warnings.
+ The '-Wdeprecated' warnings about `%pure-parser` being deprecated in the `.yy` parser files. At first I wanted to include them here, but despite things seemingly building fine it seems there's some additional implications with replacing `%pure-parser` with `%define api.pure` and `%error-verbose` with `%define parse.error verbose` that I can't judge because I'm not a bison expert.
+ Maybe additional warnings on MSVC if more than the default are turned on (`/Wall`)

yhabteab commented on lib/config/expression.cpp:
```diff
@@ -553,8 +553,9 @@ ExpressionResult GetScopeExpression::DoEvaluate(ScriptFrame& frame, DebugHint *d
-		VERIFY(!"Invalid scope.");
+
+	VERIFY(!"Invalid scope.");
+	return Empty;
```
Which compiler did complain about this? I'm not able to find this kind of warning in the GHAs logs or locally on my machine. The previous version was pretty obvious that it will never return since it's negating a statically provided string, and I'm a bit surprised that the compiler isn't able to figure this out.

jschmidt-icinga commented:
+ After switching the compiler to clang it found a few new variations of unused variables which I fixed as well. Maybe we should add a runner that also tests with clang to get those warnings too.
+ Hopefully the windows warnings about branches not returning is now fixed by making the return type of `icinga_assert_fail()` function void.
+ I've changed the timestamps used for the replay log file names from using `int` to `std::uint64_t`. I didn't just leave them as double so they can easily be converted from and to strings from/for the file names.

yhabteab reviewed (approved):
LFTM! Ty.
